test/unit/builtins/array.spec.ts

```diff
@@ -209,19 +209,11 @@ describe("delete", () => {
         `.expectToMatchJsResult();
     });
 
-    test("returns true when element exists", () => {
+    test("returns true when deletion is successful", () => {
         util.testFunction`
             const array = [1, 2, 3, 4];
-            const exists = delete array[2];
-            return { exists, a: array[0], b: array[1], c: array[2], d: array[3] };
-        `.expectToMatchJsResult();
-    });
-
-    test("returns false when element not exists", () => {
-        util.testFunction`
-            const array = [1, 2, 3, 4];
-            const exists = delete array[4];
```

Comment:
Why was this test removed? It still seems a valid thing to test.

Reply:
Both tests return `true` regardless if the element exists or not in actual JS

```
> array = [1, 2, 3, 4];
[ 1, 2, 3, 4 ]
> delete array[4];
true
```

I'm not sure of a way to delete an element from an array and expect a `false` result

Reply:
I looked into it a bit more and I've put the test back using `Object.defineProperty` to expect `success` to be `false`

Reply:
Good find

Reply:
Is this try/catch needed? It is not supposed to throw an error right?

Reply:
Nope it is meant to, you can make a `DeleteExpression` throw an error if you assign an unconfigurable descriptor using `Object.defineProperty` to the index of the array for the element you're trying to delete

Reply:
```ts
array = [1, 2, 3, 4];
Object.defineProperty(array, "0", { configurable: false });
delete array[0]; // throws an error within the test JS environment
```

Reply:
Could you split this into two tests then, one testing the error, the other testing the return value & functionality?

Reply:
f78a797 should fix this up